src/pages/lessons/[slug].tsx

```diff
@@ -228,10 +168,39 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
     summary,
   } = lesson
 
-  console.log(lesson)
+  const currentState = playerState.value
+
+  console.log(`The current player state: ${currentState}`)
+
+  React.useEffect(() => {
+    switch (currentState) {
+      case 'loading':
+        if (!isEmpty(data.lesson)) {
+          send('LOADED')
+        }
+        break
+      case 'loaded':
+        if (hls_url || dash_url) {
+          send('PLAY')
+        } else {
+          send('SUBSCRIBE')
+        }
+      case 'completed':
+        send('NEXT')
```

Comment:
when the lesson finishes we decide what overlay to display. This would also be where we would transition to a quiz, review, etc when a video completes. Currently, it's just the next lesson overlay.